cobigen-cli/src/main/java/com/devonfw/cobigen/cli/commands/CobiGenCommand.java

```diff
@@ -46,30 +50,58 @@ public void run() {
 	}
 
 	/**
-	 * This class implement getVersion() and this method return the version of plug-in
+	 * This class implement getVersion() and this method return the version of
+	 * plug-in
 	 */
 	static class PropertiesVersionProvider implements IVersionProvider {
 		@Override
 		public String[] getVersion() throws Exception {
-			String versionProvider[] = new String[50];
+			List<String> versionProvider = new ArrayList<String>();
 			MavenXpp3Reader reader = new MavenXpp3Reader();
-			Model model;
-			if ((new File("pom.xml")).exists())
-				model = reader.read(new FileReader("pom.xml"));
-			else
-				model = reader.read(
-						new InputStreamReader(CobiGenCommand.class.getResourceAsStream("/src/main/resources/pom.xml")));
+			Model model = null;
+			File locationCLI = new File(CobiGenUtils.class.getProtectionDomain().getCodeSource().getLocation().toURI());
+			Path rootCLIPath = locationCLI.getParentFile().toPath();
+
+			File pomFile = extractArtificialPom(rootCLIPath);
+
+			if (pomFile.exists()) {
+				model = reader.read(new FileReader(pomFile));
+			}
+
 			List<Model> versionList = new ArrayList<Model>();
 			versionList.add(model);
 
-			for (int j = 0; j < versionList.get(0).getDependencies().size(); j++) {
-				versionProvider[j] = " name:= " + model.getDependencies().get(j).getArtifactId() + " version=  "
-						+ model.getDependencies().get(j).getVersion();
-			}
+			List<Dependency> modelDependencies = model.getDependencies();
+
+			for (int i = 0; i < modelDependencies.size(); i++) {
+
+				versionProvider.add(" name:= " + modelDependencies.get(i).getArtifactId() + " version=  "
+						+ modelDependencies.get(i).getVersion());
 
-			return versionProvider;
+			}
+			return versionProvider.toArray(new String[versionProvider.size()]);
 		}
 
 	}
 
+	/**
+	 * Extracts an artificial POM which defines all the CobiGen plug-ins that are
+	 * needed
+	 * 
+	 * @param rootCLIPath path where the artificial POM will be extracted to
+	 * @return the extracted POM file
+	 */
+	private static File extractArtificialPom(Path rootCLIPath) {
```

Comment:
Here you are duplicating this method from CobiGenUtils. Please call directly the method from there, make it `public` if needed